LinearAlgebraForCAP/gap/DerivedMethodsForMatrixCategories.gi

```diff
@@ -0,0 +1,232 @@
+    k := CommutativeRingOfLinearCategory( cat );
+    
+    return VectorSpaceObject( 1, k );
+    
+end: ConditionsListComplete := true,
+  CategoryFilter := CATEGORY_FILTER_CHECKER,
+  Description := "Adding DistinguishedObjectOfHomomorphismStructure using external hom methods" );
+
+##
+AddFinalDerivation( HomomorphismStructureOnObjects,
+                    [
+                      [ BasisOfExternalHom, 1 ],
+                      [ CoefficientsOfMorphismWithGivenBasisOfExternalHom, 1 ],
+                      [ MultiplyWithElementOfCommutativeRingForMorphisms, 1 ] 
+                    ],
+                    [
+                      HomomorphismStructureOnObjects,
+                      HomomorphismStructureOnMorphismsWithGivenObjects,
+                      DistinguishedObjectOfHomomorphismStructure,
+                      InterpretMorphismAsMorphismFromDistinguishedObjectToHomomorphismStructure,
+                      InterpretMorphismFromDistinguishedObjectToHomomorphismStructureAsMorphism
+                    ],
+  
+  function( a, b )
+    local cat, k, dimension;
+    
+    cat := CapCategory( a );
+    
+    k := CommutativeRingOfLinearCategory( cat );
+    
+    dimension := Length( BasisOfExternalHom( a, b ) );
+    
+    return VectorSpaceObject( dimension, k );
+    
+end: ConditionsListComplete := true,
+  CategoryFilter := CATEGORY_FILTER_CHECKER,
+  Description := "Adding HomomorphismStructureOnObjects using external hom methods" );
+
+##
+AddFinalDerivation( HomomorphismStructureOnMorphismsWithGivenObjects,
+                    [
+                      [ BasisOfExternalHom, 1 ],
+                      [ CoefficientsOfMorphismWithGivenBasisOfExternalHom, 1 ],
+                      [ MultiplyWithElementOfCommutativeRingForMorphisms, 1 ],
+                      [ PreCompose, 1 ]
+                    ],
+                    [
+                      HomomorphismStructureOnObjects,
+                      HomomorphismStructureOnMorphismsWithGivenObjects,
+                      DistinguishedObjectOfHomomorphismStructure,
+                      InterpretMorphismAsMorphismFromDistinguishedObjectToHomomorphismStructure,
+                      InterpretMorphismFromDistinguishedObjectToHomomorphismStructureAsMorphism
+                    ],
+  
+  #          alpha
+  #      a --------> a'     s = H(a',b) ---??--> r = H(a,b')
+  #      |           |
+  # alpha.h.beta     h
+  #      |           |
+  #      v           v
+  #      b' <------- b
+  #          beta
+  
+  function( hom_source, alpha, beta, hom_range )
+    local cat, k, dim_hom_source, dim_hom_range, B, mat;
+    
+    cat := CapCategory( alpha );
+    
+    k := CommutativeRingOfLinearCategory( cat );
+    
+    dim_hom_source := Dimension( hom_source );
+    
+    dim_hom_range := Dimension( hom_range );
+    
+    if dim_hom_source * dim_hom_range = 0 then
+      
+      return ValueGlobal( "ZeroMorphism" )( hom_source, hom_range ); # sanity checks for final derivations
```

Comment:
Same happens here, there should be a way to let sanity checks (for final derivation) to omit some operations, if they happen in other categories.

Reply:
@sebastianpos